### ISSUE TITLE
support menues

### ISSUE THREAD

**Issue by flosse** [cobalt-org/cobalt.rs#89]
It would be great to be able to define a menu within the configuration file.

**Comment by LucioFranco** [comment]
What type of menu are you talking about? You can create your own menu with css. Are you talking about adding our own menus that you can just enable?

**Comment by flosse** [comment]
I'd like to define a menu structure like this

```yaml
menu:
  - name: "Home"
    url: "/"
  - name: "About"
    url: "/about/"
```
to use it within a navigation template.

**Comment by LucioFranco** [comment]
Seems like an interesting idea. I did some research and found [this](https://github.com/Ahti/jekyll-menu#usage). Seems like this might be a better idea to do with a plugin system. I don't think this is something that should be added to the overall cobalt project.

**Comment by flosse** [comment]
> Seems like this might be a better idea to do with a plugin system

Do you know a rust project with a simple plugin system?
Might be related to #57.

**Comment by LucioFranco** [comment]
I think there could be two paths for it. One is to use compiler features or possibly do something like cargo subcommands. `cobalt-menu` and in the config you add it and cobalt will go and look for it at runtime. You could also execute the command manually. What do you think?

**Comment by flosse** [comment]
> One is to use compiler features

Are you talking about the cargo features flag ( e.g. `cargo build --features menu`) ?
That seems to be a simple way to go :)

>  or possibly do something like cargo subcommands. 

I have no idea how they are working :-\

> `cobalt-menu` and in the config you add it and cobalt will go and look for it at runtime

sounds good :)

> You could also execute the command manually. What do you think?

I don't want to execute sub-commands manually for each plugin on every build ;-)

**Comment by LucioFranco** [comment]
@flosse So the idea would be that you `cargo install cobalt-menus` and you would specify that you want cobalt to execute that plugin when you execute `cobalt build`. You can also execute the menu plugin by calling `cargo-menus`. Does that make sense? I think that this might be the best approach. Would love your input @johannhof 😄

**Comment by jespino** [comment]
Can be interesting to add global-context variables in the cobalt.yaml (This solve the problem and can have sense in the core of cobalt).

For example, I can want to define the contact address, the title of the page or other shared data between a lot of pages. This is interesting too for the possibility of templates, because you can distribute a template with a set of variables to define.

**Comment by epage** [comment]
I wonder if one way of resolving this is to implement support for [data files](https://jekyllrb.com/docs/datafiles/) like I suggested in #91.  I think it'd be a bit cleaner than adding site variables like in #171 or defaults like what might be done as part of #171 or #199.

**Comment by epage** [comment]
Agreed but would like to wait to see if any of the other people on this thread agree.

**Comment by epage** [comment]
Closing as fixed by #256 unless the filer would like to contest that.
